tests/testSphPoint.cc

```diff
@@ -0,0 +1,69 @@
+// -*- LSST-C++ -*-
+
+/*
+ * LSST Data Management System
+ * See COPYRIGHT file at the top of the source tree.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program. If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include <iostream>
+#include <sstream>
+
+// FIXME: violates https://developer.lsst.io/coding/unit_test_policy.html#testing-frameworks
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE SphPointCpp
+
+#include "boost/test/unit_test.hpp"
+
+#include "lsst/afw/geom/SphPoint.h"
+
+/*
+ * Unit tests for C++-only functionality in SphPoint.
+ *
+ * See testSphPoint.py for remaining unit tests.
+ */
+namespace lsst {
+namespace afw {
+namespace geom {
+
+    /**
+     * @brief Tests whether the result of SphPoint::SphPoint(SphPoint const &)
+     * makes an identical but independent copy.
+     */
+    BOOST_AUTO_TEST_CASE(SphPointCopyResult, *boost::unit_test::tolerance(1e-10))
+    {
+        SphPoint original(Point3D(0.0, 0.0, 1.0));
```

Comment:
I'm surprised you picked a point at the pole, since the equality operator ignores longitude there. A different point would give slightly broader coverage.